### PR TITLE
Add elements not found to contingency

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/actions/server/dto/ContingencyInfos.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.gridsuite.actions.server.dto;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/actions/server/dto/ContingencyInfos.java:1]: Add license and author

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 2]
```diff
@@ -252,20 +280,36 @@ public ScriptContingencyList newScriptFromFormContingencyList(UUID id, UUID newI
...
-    private static IdBasedContingencyList fromIdBasedContingencyListEntity(IdBasedContingencyListEntity entity) {
+    private static IdBasedContingencyList fromIdBasedContingencyListEntity(IdBasedContingencyListEntity entity, Network network) {
         List<NetworkElementIdentifier> listOfNetworkElementIdentifierList = new ArrayList<>();
+        Map<String, Set<String>> notFoundElements = new HashMap<>();
         entity.getIdentifiersListEntities().forEach(identifierList -> {
             List<NetworkElementIdentifier> networkElementIdentifiers = new ArrayList<>();
-            identifierList.getEquipmentIds().forEach(equipmentId -> networkElementIdentifiers.add(new IdBasedNetworkElementIdentifier(equipmentId)));
+            identifierList.getEquipmentIds().forEach(equipmentId -> {
+                if (network != null && network.getIdentifiable(equipmentId) == null) {
+                    Set<String> ids = notFoundElements.get(identifierList.getName());
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:290]: Use Map.computeIfAbsent to be more concise ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/actions/server/ContingencyListService.java]: Done

File: src/main/java/org/gridsuite/actions/server/dto/ContingencyInfos.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/actions/server/dto/ContingencyInfos.java:8]: Done

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 3]
```diff
@@ -132,12 +132,47 @@ public Optional<PersistentContingencyList> getFormContingencyList(UUID id) {
...
+    }
+
+    Optional<List<ContingencyInfos>> exportContingencyList(UUID id, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(id);
+        Network network = getNetworkFromUuid(networkUuid, variantId);
+        Optional<PersistentContingencyList> persistentContingencyList = getScriptContingencyList(id)
+                .or(() -> getFormContingencyList(id))
+                .or(() -> getIdBasedContingencyList(id, network));
+
+        Optional<List<Contingency>> contingencies = getScriptContingencyList(id).map(contingencyList -> getPowsyblContingencies(contingencyList, network))
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:152]: Optional<List<Contingency>> contingencies = persistentContingencyList.map(contingencyList -> getPowsyblContingencies(contingencyList, network));

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 4]
```diff
@@ -132,12 +132,47 @@ public Optional<PersistentContingencyList> getFormContingencyList(UUID id) {
...
-    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, UUID networkUuid, String variantId) {
+    private List<Contingency> getPowsyblContingencies(PersistentContingencyList contingencyList, Network network) {
+        ContingencyList powsyblContingencyList = contingencyList.toPowsyblContingencyList(network);
+        return powsyblContingencyList == null ? Collections.emptyList() : powsyblContingencyList.getContingencies(network);
+    }
+
+    Optional<List<ContingencyInfos>> exportContingencyList(UUID id, UUID networkUuid, String variantId) {
+        Objects.requireNonNull(id);
+        Network network = getNetworkFromUuid(networkUuid, variantId);
+        Optional<PersistentContingencyList> persistentContingencyList = getScriptContingencyList(id)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:148]: You could get rid of the Optional here so you don't need to deal with it in the whole method

File: src/main/java/org/gridsuite/actions/server/ContingencyListService.java [hunk 5]
```diff
@@ -132,12 +132,47 @@ public Optional<PersistentContingencyList> getFormContingencyList(UUID id) {
...
+                .or(() -> getFormContingencyList(id))
+                .or(() -> getIdBasedContingencyList(id, network));
+
+        Optional<List<Contingency>> contingencies = getScriptContingencyList(id).map(contingencyList -> getPowsyblContingencies(contingencyList, network))
+            .or(() -> getFormContingencyList(id).map(contingencyList -> getPowsyblContingencies(contingencyList, network)))
+                    .or(() -> getIdBasedContingencyList(id, network).map(contingencyList -> getPowsyblContingencies(contingencyList, network)));
+
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/actions/server/ContingencyListService.java:155]: You should declare a List<ContingencyInfos>> contingencyInfos; 
And then you fill it with something like 
add(new ContingencyInfos(stringSetEntry.getKey(), null, stringSetEntry.getValue()))) 
or 
add(new ContingencyInfos(contingency, notFoundElements.get(contingency.getId()))) 
and then you return the list.
The code is a bit difficult here

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**